_rules/aria-state-or-property-valid-value-6a7281.md

```diff
@@ -88,6 +88,7 @@ Some user agents treat the value of `aria-*` attribute as case-sensitive (even w
 
 ```html
 <div role="textbox" aria-errormessage="my-error" aria-label="A textbox"></div>
+<div id="my-error">My error message</div>
```

Comment:
This is not required by the rule, see note 1 in the expectation:

> Only for [WAI-ARIA required properties](https://www.w3.org/TR/wai-aria-1.1/#requiredState) with value types `ID Reference` and `ID Reference List` is there a requirement that the elements with the given ids actually exists. For non-required properties, this is not a requirement.

An example of what might happen with aria-errormessage is that the aria-errormessage attribute is static, and including or removing the element with the ID is part of setting the error. Something like this:

```html
<template>
  <input aria-errormessage="err1" :aria-invalid="error ? true : false">
  <p v-if="error" id="err1">{{ error }}<p>
</template>
```

I've seen this in practice too. There is no difference between this, and actually removing the aria-errormessage attribute. The way this is written using a note isn't very clear, and will need to be improved.

Reply:
I opened #1617 because the testcase does not indicate that as a reasoning. It says:
> aria-errormessage property with valid ID reference value

That's misleading if that's the reason it's passing. So, I was asked to add a PR to add the reference rather than change the message.

On our end, we'll probably still end up flagging it anyway since allowing missing references will also mask mistaken references. If you have aria-errormessage pointing at er1 and the actual id is err1, the developer would never notice it, even though they think they did the right thing. In your example, it's just as easy to do something like `error ? "messageId": undefined`.

Reply:
Yeah, I agree with you, this isn't worded as well as it could be. It also shouldn't be in a note, it needs to be part of the expectation text itself. It's too easy to miss.

Would it help your case if we added aria-invalid=false on this example, to have a more explicit clue about what is going on? If so, feel free to make that change.

Reply:
Honestly, if that's the intent, I would just change the description from:
> aria-errormessage property with valid ID reference value

to:

> aria-errormessage property with a non-existent ID reference value, but aria-errormessage is not required. See expectation note.

And not mess with changing the actual testcase. Does that make sense?

Reply:
Hi @tombrunet, we've discussed that on today call and indeed agreed that updating the description of the example is what makes sense 👍 
The example (and Failed Example 4) is more or less here to point out that "valid ID" doesn't mean "existing" but "OK according to ID parsing rules".

We can also add a link (from the expectation) to the HTML spec explaining that. https://html.spec.whatwg.org/multipage/dom.html#the-id-attribute
> When specified on HTML elements, the id attribute value must be unique amongst all the IDs in the element's tree and must contain at least one character. The value must not contain any ASCII whitespace.

Can you handle that?